connectors/sources/dropbox.py

```diff
@@ -164,8 +250,130 @@ async def ping(self):
             )
         except AuthError:
             raise ConfigurableFieldValueError("Configured Refresh Token is invalid")
+        except Exception:
+            raise
+
+    async def get_content(self, attachment, response=None, timestamp=None, doit=False):
+        """Extracts the content for allowed file types.
+
+        Args:
+            attachment (dictionary): Formatted attachment document.
+            response (object): Response of a shared file
+            timestamp (timestamp, optional): Timestamp of attachment last modified. Defaults to None.
+            doit (boolean, optional): Boolean value for whether to get content or not. Defaults to False.
+
+        Returns:
+            dictionary: Content document with _id, _timestamp and attachment content
+        """
+        attachment_size = int(attachment.size)
+        if not (doit and attachment_size > 0):
+            return
+
+        attachment_name = attachment.name
+        attachment_extension = os.path.splitext(attachment_name)[-1]
+
+        if attachment_extension == "":
+            logger.warning(
+                f"Files without extension are not supported by TIKA, skipping {attachment_name}."
+            )
+            return
+
+        elif attachment_extension not in TIKA_SUPPORTED_FILETYPES:
+            logger.warning(
+                f"Files with the extension {attachment_extension} are not supported by TIKA, skipping {attachment_name}."
+            )
+            return
+
+        if attachment_size > FILE_SIZE_LIMIT:
+            logger.warning(
+                f"File size {attachment_size} of file {attachment_name} is larger than {FILE_SIZE_LIMIT} bytes. Discarding file content"
+            )
+            return
+
+        logger.debug(f"Downloading {attachment_name}")
+
+        document = {
+            "_id": attachment.id,
+            "_timestamp": attachment.server_modified,
+        }
+        temp_filename = ""
+        async with NamedTemporaryFile(mode="wb", delete=False) as async_buffer:
```

Comment:
`get_content` is doing too much from my point of view. I think this block of code can be defined in its own function.

Reply:
Yes, this block of code can be moved to another function.